### PR TITLE
feat: migrate Segment main app files [split of 8387 - 3]

### PR DESCRIPTION
**Pull request by NicolasMassart**
## **Description**

>[!NOTE]
> #8387 is too big for review, splitting it into contextual/domain scoped smaller PRs.
> you can refer to #8387 for global view

> [!WARNING]
> This PR depends on PR slice 2 #8574, currently targeting `feat/1461_migrate_segment_utils_2`  branch to allow build. Will require change once branch merged in `feat/batch_1129_segment`

### This is the part of #8387 related to the utils code of the Segment work.

update main files to use metrics instance

This also requires updating of some components and tests because they indirectly depend on `app/util/confirmation/signatureUtils.js` that is tested here out of the component under test
scope.
This makes the test fail when there's no call to the MetaMetrics class in the MessageSign comp. Updating test to keep coverage.
But eventually this is the best example of what we should refactor in a unit test.

## **Related issues**

Related to MetaMask/mobile-planning/issues/1461

## **Manual testing steps**

NA

### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/core/Engine.ts [hunk 1]
```diff
@@ -861,13 +860,16 @@ class Engine {
...
           // Track detected tokens event
-          AnalyticsV2.trackEvent(MetaMetricsEvents.TOKEN_DETECTED, {
-            token_standard: 'ERC20',
-            asset_type: 'token',
-            chain_id: getDecimalChainId(
-              networkController.state.providerConfig.chainId,
-            ),
-          });
+          MetaMetrics.getInstance().trackEvent(
+            MetaMetricsEvents.TOKEN_DETECTED.category,
```
Comment (Cal-L) [inline app/core/Engine.ts:864]: Did the first parameter change? This would change it from an object to a string instead.

Comment (Cal-L) [reply app/core/Engine.ts:864]: Same question for the other events

**Review by Cal-L — COMMENTED** [review]
Left a comment

File: app/core/Engine.ts (hunk 1 above)
Comment (NicolasMassart) [reply app/core/Engine.ts:864]: Yes, that's a small discrepency I detected too.
I started with string in the class but then found it was easier to have the object in the hook as it's called far more often. I should change the class type too. Will do. Thanks for the reminder.

Comment (NicolasMassart) [reply app/core/Engine.ts:864]: Will do in a separate PR though

Comment (Cal-L) [reply app/core/Engine.ts:865]: Sorry I wasn't following regarding the `object in the hook` and `change the class type`. To keep this consistent, could we just remove the `.category` here or did the `trackEvent` parsing logic change?

**Review by Cal-L — COMMENTED** [review]
Left a comment

**Comment by NicolasMassart** [comment]
Closed as this was reworked and split differently in smaller PRs for the migration

**Pull request closed without merging**